### PR TITLE
Fuzzy match the role command.

### LINKED ISSUE #1194: Add fuzzy matching to the role(s) command.
Right now, if you want to use the role name for the `!role(s)` commands, you have to type the *exact* role name, which can be quite frustrating sometimes. 

It would be nice to support fuzzy matching, like we do here with the `Tags` cog.

https://github.com/python-discord/bot/blob/9e95fb0a34733d6ce9d9619e892a860a54567606/bot/exts/info/tags.py#L69-L83

### PR DESCRIPTION
**Pull request by kosayoda**
Resolves #1194.

- The `!role` command now fuzzy matches.
image

- Duplicates are filtered.

### CODE REVIEW

**Review by MarkKoz — APPROVED** [review]
It'd be nice for it to work on individual words e.g. `events` for `events team` without just reducing the threshold. That's probably too complicated. This is still fine.

**Review by Den4200 — APPROVED** [review]
Everything looks and works fine!

The only thing is the output of roles are not in the given order because of the switch to sets. It's nothing major, but I'll let you decide if we want to keep the sets or switch back to lists.

**Review by kwzrd — APPROVED** [review]
Works just fine; have you considered also trying a direct substring search? That would allow for e.g. `"Core Dev"` to match, but it'd only make sense if the search term is n chars long. Perhaps needlessly complex, as Mark says.

File: bot/exts/info/information.py
Comment (kosayoda) [reply bot/exts/info/information.py:129]: > something to do with matching against a dict

yep!

> Out of interest, is this documented somewhere?

It's a bit of a dig. The documentation for [`extractOne`](https://github.com/seatgeek/fuzzywuzzy/blob/2188520502b86375cf2610b5100a56935417671f/fuzzywuzzy/process.py#L197) which is used does only say a tuple containing the match and the score, but it uses [`extractWithoutOrder`](https://github.com/seatgeek/fuzzywuzzy/blob/2188520502b86375cf2610b5100a56935417671f/fuzzywuzzy/process.py#L16) internally, which [documents the 3-tuple return value here](https://github.com/seatgeek/fuzzywuzzy/blob/2188520502b86375cf2610b5100a56935417671f/fuzzywuzzy/process.py#L51-L60)

Comment (kwzrd) [reply bot/exts/info/information.py:129]: Interesting, thanks for the explanation.

**Comment by Xithrius** [comment]
@kosayoda we will be able to merge after you fix bot building errors.

**Comment by kosayoda** [comment]
sigh forgot to pull master before merging, at least it's not conflict inducing

**Pull request merged**
